generators/csharp/sdk/src/oauth/OauthTokenProviderGenerator.ts

```diff
@@ -0,0 +1,273 @@
+                })
+            );
+
+            writer.writeNodeStatement(
+                csharp.codeblock((writer) => {
+                    const requestType = this.tokenEndpoint.requestBody?._visit({
+                        reference: (value) => {
+                            return this.context.csharpTypeMapper.convert({ reference: value.requestBodyType });
+                        },
+                        inlinedRequestBody: (value) => {
+                            return csharp.Type.reference(
+                                this.context.getRequestWrapperReference(
+                                    this.tokenEndpointReference.serviceId,
+                                    value.name
+                                )
+                            );
+                        },
+                        fileUpload: (value) => undefined,
+                        bytes: (value) => undefined,
+                        _other: (value) => undefined
+                    });
+                    if (requestType == null || requestType.internalType.type !== "reference") {
+                        throw new Error("Failed to get request class reference");
+                    }
+
+                    writer.write("var tokenResponse = ");
+                    writer.writeNode(
+                        csharp.invokeMethod({
+                            async: true,
+                            method: `client.${this.tokenEndpoint.name.pascalCase.safeName}`,
+                            arguments_: [
+                                csharp.instantiateClass({
+                                    classReference: requestType.internalType.value,
+                                    // TODO(dsinghvi): assumes only top level client id and client secret inputs
+                                    arguments_: [
+                                        {
+                                            name: this.context.getNameForField(
+                                                this.scheme.configuration.tokenEndpoint.requestProperties.clientSecret
+                                                    .property.name
+                                            ),
+                                            assignment: csharp.codeblock(
+                                                OauthTokenProviderGenerator.CLIENT_ID_FIELD.name
+                                            )
+                                        },
+                                        {
+                                            name: this.context.getNameForField(
+                                                this.scheme.configuration.tokenEndpoint.requestProperties.clientSecret
+                                                    .property.name
+                                            ),
+                                            assignment: csharp.codeblock(
+                                                OauthTokenProviderGenerator.CLIENT_ID_FIELD.name
```

Comment:
I think this should be:

```suggestion
                                                OauthTokenProviderGenerator.CLIENT_SECRET_FIELD.name
```